src/index.js

```diff
@@ -2,13 +2,29 @@
 
 const coWrapper = require('./lib/coWrapper')
 const processImage = require('./lib/processImage')
+const loaderUtils = require('loader-utils')
 
 const loader = coWrapper(function* (content) {
-  if (typeof this.query === 'string') {
-    throw new Error('does not support inline querystring as options, define your options in webpack.config.js instead')
-  }
+  const globalOptions = loaderUtils.getOptions(this);
+  const resourceOptions = loaderUtils.parseQuery(this.resourceQuery || "?")
+  let presetOptionsList = [];
 
-  const sharpInstance = processImage(content, this.query)
+  // We have 2 possible choices:
+  // - set of presets in `presets`
+  // - single preset in `preset`
+  if (resourceOptions.preset) {
+    // @todo duplicate code
+    presetOptionsList = [resourceOptions.preset].map((key) => (globalOptions.presets[key]))
+  }
+  else if (Array.isArray[resourceOptions.presets]) {
+    // @todo duplicate code
+    presetOptionsList = resourceOptions.presets.map((key) => (globalOptions.presets[key]))
+  }
+  const options = Object.assign({}, globalOptions, ...presetOptionsList, resourceOptions)
```

Comment:
it is a shallow merge, which means, from your example in README
preset=blur will overwrite `jpeg: { progressive: true }` by `jpeg: { quality: 55 }`
I think user will expect `jpeg: { progressive: true, quality: 55 }`, do you agree?

I think we can use sth like http://ramdajs.com/docs/#mergeDeepRight to help merge
but not lodash merge because it will concat array